src/cli/serve/read_yaml_config.js

```diff
@@ -38,38 +43,40 @@ const deprecatedSettings = {
   'server.xsrf.token': 'server.xsrf.token is deprecated. It is no longer used when providing xsrf protection.'
 };
 
-module.exports = function (path) {
-  if (!path) return {};
-
-  let file = yaml.safeLoad(fs.readFileSync(path, 'utf8'));
-
-  function apply(config, val, key) {
-    if (_.isPlainObject(val)) {
-      _.forOwn(val, function (subVal, subKey) {
-        apply(config, subVal, key + '.' + subKey);
-      });
-    }
-    else if (_.isArray(val)) {
-      config[key] = [];
-      val.forEach((subVal, i) => {
-        apply(config, subVal, key + '.' + i);
-      });
-    }
-    else {
-      _.set(config, key, val);
-    }
-  }
+export default function (paths) {
+  if (!paths) return {};
 
-  _.each(deprecatedSettings, function (message, setting) {
-    if (_.has(file, setting)) console.error(message);
+  const files = [].concat(paths).map(path => {
+    return safeLoad(read(path, 'utf8'));
   });
 
-  // transform legeacy options into new namespaced versions
-  return _.transform(file, function (config, val, key) {
-    if (legacySettingMap.hasOwnProperty(key)) {
-      key = legacySettingMap[key];
-    }
+  return _.transform(files, function applyConfigFile(config, file) {
+    _.forOwn(file, function apply(val, key) {
```

Comment:
Moving the apply() definition into this loop and changing it so it closes over `config` and `legacySettingMap` makes this code more complex.  The original function wasn't pure since it mutated the given config, but it did at least only act upon and with the arguments it was given.

I think it may add a lot of clarity and improve testability of this code if we were to move the "legacy settings" logic into its own module rather than coupling it to the "read config" module.

The existing module had a lot of concerns spread throughout procedural code, so rather than compounding on that, this seems like a good opportunity to fundamentally improve it.